Export ignore

lippserd opened the pull request:
The main topic of this PR is removing scripts, RELEASE.md and Git-related files from release tarballs.
It also includes minor changes to the make-release script.

Al2Klimov commented:
1. Does 6eed5dc cover the four bullet points?
2. Didn't `git bisect` make tracking *everything* as easy as 1-2-3? If not: again, 6eed5dc?

Al2Klimov commented:
@lippserd IMO it's not as performance-wasting as we've discussed (offline) – even @Thomas-Gelf [says](https://github.com/Icinga/ipl/pull/5#pullrequestreview-91107806) "Looks good to me." Everything is passed by reference and only touched if it's on object.

Al2Klimov commented:
Rebased

lippserd commented:
Just for the records, the tests are still missing

lippserd committed:
Streamline .gitignore

lippserd committed:
Silence vendor directory removal

With the --ignore-unmatch, the make-release script no longer complains
that the path specification does not match any file the first time it
is called. All subsequent calls to the make-release script no longer
print the removed files with the -q switch.

nilmerg commented on bin/make-release.sh:
```diff
@@ -43,10 +43,16 @@ find vendor/ -type f -name "*.php" \
 find vendor/ -type f -name LICENSE | xargs -L1 git add -f
 find vendor/ -type f -name '*.json' | xargs -L1 git add -f
-find asset/ -type f | xargs -L1 git add -f
+find -L asset/ -type f | xargs -L1 git add -f
```
Why is this necessary? The assets are copied over, not symlinked.

lippserd replied:
You're right! Maybe I was using an older version with this change?

nilmerg commented on bin/make-release.sh:
```diff
@@ -43,10 +43,16 @@ find vendor/ -type f -name "*.php" \
+bin/ export-ignore
+RELEASE.md export-ignore
+gitattributes
+git add .gitattributes
```
Why is this not a file in git?

lippserd replied:
I thought it was okay to still have them when archives are being created from the master branch.

nilmerg replied:
Why should anyone do so? The readme states that the master branch is not usable. And if one wants to create a new release, publishing it requires git access anyway.
